### PR TITLE
Genesis block

### PR DESCRIPTION
**Pull request by seuushi**
Implement genesis block.

When a block is going to be added to the blockchain, it first checks whether the blockchain already contains blocks. If the blockchain does not contain any blocks, a genesis block will be added first. The genesis block is the very first block in the blockchain.

### CODE REVIEW

File: market/database/backends.py [hunk 1]
```diff
@@ -465,6 +477,50 @@ def get_latest_sequence_number(self):
...
+        :param public_key_benefactor: The public key of the benefactor
+        :param public_key_beneficiary The public key of the beneficiary
+        """
+        db_query = u"SELECT COUNT(*) FROM multi_chain"
+        db_result = self.execute(db_query).fetchone()
+
+        if db_result[0] == 0:
+            insert_time = int(time.time())
+            # Hash the block
+            packet = encode(
```
Comment (Jumba) [inline market/database/backends.py]: Is it possible to actually create a `DatabaseBlock' that is the genesis block? And make a seperate function to generate the block and return it. Then you can just do `BlockChain.add_block(genesis)' right?

File: test/market/test_multichain.py [hunk 2]
```diff
@@ -246,8 +247,40 @@ def test_update_block_with_beneficiary(self):
...
 
+    def test_check_add_genesis_block(self):
+        """
+        This test checks the functionality of adding a block to an empty blockchain.
+        """
+
+        # Add genesis block to the blockchain
+        self.db.check_add_genesis_block(self.member_bank, '')
+        # Get the genesis block by the public key and sequence number
+        genesis_block = self.db.get_by_public_key_and_sequence_number(str(self.member_bank), 0)
```
Comment (Jumba) [inline test/market/test_multichain.py]: If you add the ability to create a genesis block separately you can check the "check_add_genesis_block" step, and just create a genesis block. Which you can use below to check if the new block that has been added automatically triggered the genesis block creation by checking if the previous block hash is equal to the genesis hash.

File: market/database/backends.py (hunk 1 above)
Comment (seuushi) [inline market/database/backends.py]: Pushed the changes

File: test/market/test_multichain.py (hunk 2 above)
Comment (seuushi) [inline test/market/test_multichain.py]: Not sure what you meant with this. Please check the updated code.

File: market/database/backends.py [hunk 3]
```diff
@@ -153,9 +153,26 @@ def get_latest_sequence_number(self):
...
+    def get_next_sequence_number(self):
         """
-        Persist the genesis block if there are no blocks yet.
+        Return the next sequence number for this public_key.
+        If no block for the pk is known return 0, else return latest sequence number + 1.
+        :return: sequence number (integer)
+        """
+        raise NotImplementedError
+
+    def create_genesis_block(self, public_key_benefactor, public_key_beneficiary):
```
Comment (Jumba) [inline market/database/backends.py]: I think the genesis block should be equal over all clients.

**Review by Jumba — CHANGES_REQUESTED** [review]

File: market/database/backends.py (hunk 3 above)
Comment (seuushi) [inline market/database/backends.py]: Review again please.

**Review by Jumba — APPROVED** [review]

**Review by tanthanadon — COMMENTED** [review]
Helllo

**Pull request merged**